### ISSUE TITLE
Payment logs: if we can't record the payment the log is for, at least record the transaction

### ISSUE THREAD

**Issue by sethshoultes** [eventespresso/event-espresso-core#544]
**Codebase ticket:** 
https://events.codebasehq.com/projects/event-espresso/tickets/11367

**Testing notes:**
To test this use master and the iPay88 payment method (it doesn't need to be set up to function, just add M03222 as the merchant code and Tmnx7fmSu4 as the merchant key)

The reason for using that payment method is right now it creates logs that use the transaction object rather the the payment object.

Setup a paid ticket and register onto it using iPay88.

Check the PM logs and you'll find 2 of them do not link to a transaction - http://take.ms/9Q0SK

Now switch to BUG/11361/fix-log-calls-using-the-transaction-object-rather-than-the-payment-object

Repeat the same test and check the payment method logs, this time you'll see the logs link to the transaction, but show a payment method of unknown - http://take.ms/LpjnU

Confirm that the payment method in both the unknown logs and the 'normal' logs show correctly when you view the log itself:
Unknown - http://take.ms/aR8rW
Normal - http://take.ms/Ec0zV

There will be another ticket to improve this but for now its better to know the logs are for a specific transaction and work from there, than to know theres 'a' log entry for a payment method, but not know what its for.

The above payment will fail, now select a payment method to finalize the registration (complete it using another like Stripe if preferred).

Confirm the logs still show correctly for the iPay88 payments and the new one.

**Comment by sethshoultes** [comment]
@Pebblo does this still need to be tested? I see a related Codebase ticket that still needs to be merged:
https://events.codebasehq.com/projects/event-espresso/tickets/11361

**Comment by Pebblo** [comment]
Yes, this still needs to be tested, but I got multiple conflicts on when merging master so I've created a new branch with the changes.

WRT - https://events.codebasehq.com/projects/event-espresso/tickets/11361

That's for the iPay88 payment method and the reason I haven't merged it in yet is to help with testing this branch. The iPay88 payment method uses the transaction object for all log calls currently, so its really simple to test this change to confirm it works.

It's 'better' to use the payment object for the logs which is what that ticket changes in iPay88.
